### PR TITLE
Fix wrong ov datamodule update in predict, test, optimize

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

Fix https://jira.devtools.intel.com/browse/CVS-135468

Found that changing the OV test pipeline was causing a behavior that changed not only the transforms, but also other data-related configs, so fix around it.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by jaegukhyun — DISMISSED** [review]

File: src/otx/engine/engine.py [hunk 1]
```diff
@@ -533,7 +533,7 @@ def optimize(
 
         model = self.model
         if not isinstance(model, OVModel):
-            datamodule = self._auto_configurator.get_ov_datamodule()
+            optimize_datamodule = self._auto_configurator.update_ov_test_pipeline(datamodule=optimize_datamodule)
```
Comment (vinnamkim) [inline src/otx/engine/engine.py]: It is basically not allowed to use a test set for model optimization. Anyway, why do we have to create a new datamodule even if we already has `self.datamodule`?

Comment (harimkang) [reply src/otx/engine/engine.py]: This is the code added by https://github.com/openvinotoolkit/training_extensions/pull/3028 in that PR. 
@sovrasov Could you please answer this question?

Comment (harimkang) [reply src/otx/engine/engine.py]: I removed this line to update the `datamodule` in optimize. - https://github.com/openvinotoolkit/training_extensions/pull/3139/commits/edadbb6434ecfe51b0d17c3c901f098b09817589

Comment (vinnamkim) [reply src/otx/engine/engine.py]: > Anyway, why do we have to create a new datamodule even if we already has self.datamodule?

This is not just for the optimize. Why do we need these recreations for others?

Comment (harimkang) [reply src/otx/engine/engine.py]: I don't know the details, but there was a request from Vlad to update the test pipeline use the pipeline in `openvino_model.yaml` when using `OVModel`. - https://github.com/openvinotoolkit/training_extensions/pull/2897#issuecomment-1931862841

Comment (vinnamkim) [reply src/otx/engine/engine.py]: I see the reason. Can you update this to update the train pipeline with the OV data pipeline such as

```python
# not update_ov_test_pipeline()
optimize_datamodule = self._auto_configurator.update_ov_train_pipeline(datamodule=optimize_datamodule)
```

Comment (harimkang) [reply src/otx/engine/engine.py]: I modified the `update_ov_test_pipeline` function to `update_ov_subset_pipeline` to allow it to be updated by specifying a subset. - https://github.com/openvinotoolkit/training_extensions/pull/3139/commits/13ac06895c230a7c02312ae757d62b95b68c906b

**Review by jaegukhyun — APPROVED** [review]

**Review by vinnamkim — APPROVED** [review]

**Pull request merged**